### PR TITLE
Libnetconf2 build fixes for curiOS

### PR DESCRIPTION
**Pull request by troglobit**
While upgrading curiOS to Buildroot v2024.02-kkit I found that libcurl was not built properly.  The libcurl dependency is relatively new and it also needs openssl, neither of which are selected by any defconfig in curiOS.  So it was great test of our migration of select/deps from netopeer2 to libnetconf2 -- turns out a few minor things were missing, in particular the following two lines in `libnetconf2/Config.in`:

	select BR2_PACKAGE_LIBCURL # For TLS
	select BR2_PACKAGE_LIBCURL_OPENSSL # For TLS

It should work as-is now in both Infix and curiOS, but I've not tested the former.

### CODE REVIEW

File: package/libnetconf2/libnetconf2.mk [hunk 1]
```diff
@@ -9,15 +9,15 @@ LIBNETCONF2_SITE = $(call github,CESNET,libnetconf2,v$(LIBNETCONF2_VERSION))
 LIBNETCONF2_INSTALL_STAGING = YES
 LIBNETCONF2_LICENSE = BSD-3-Clause
 LIBNETCONF2_LICENSE_FILES = LICENSE
-LIBNETCONF2_DEPENDENCIES = libyang libcurl
-HOST_LIBNETCONF2_DEPENDENCIES = host-libyang host-libcurl
+LIBNETCONF2_DEPENDENCIES = libyang libopenssl libcurl
```
Comment (mattiaswal) [inline package/libnetconf2/libnetconf2.mk:12]: Missing libssh?

**Review by mattiaswal — CHANGES_REQUESTED** [review]

File: package/libnetconf2/libnetconf2.mk (hunk 1 above)
Comment (troglobit) [reply package/libnetconf2/libnetconf2.mk:12]: Yup, thanks!!  This is why I looove reviews

Comment (troglobit) [reply package/libnetconf2/libnetconf2.mk:12]: Turns out that dep was added further down in the .mk file already.  But I found another thing in the Config.in: seems we should select `BR2_PACKAGE_LIBOPENSSL` and not `BR2_PACKAGE_OPENSSL`, the latter seems to be the meta package we talked about earlier.

Fixing that up now.

**Review by mattiaswal — APPROVED** [review]
:+1:

**Pull request merged**